Fatras/include/ActsFatras/Kernel/Simulator.hpp

```diff
@@ -67,17 +77,18 @@ struct ParticleSimulator {
   ///
   /// @tparam generator_t is the type of the random number generator
   template <typename generator_t>
-  Acts::Result<InteractorResult> simulate(
+  Acts::Result<detail::InteractorResult> simulate(
```

Comment:
As this is a public method which returns the type your moving into `detail::`. Therefore, `InteractorResult` **is** part of the public interface, and should **not** be in `detail::`.

Reply:
Yes, you are right. I updated the PR. The renamed `SimulationResult` is now part of the public namespace while the `Interactor` is in the private one.

I also added a drive-by fix for an incorrect member variable name in the selectors.